galaxy_ng/app/api/v1/tasks.py

```diff
@@ -76,12 +90,154 @@ def find_real_role(github_user, github_repo):
     return real_role, real_namespace_name, real_github_user, real_github_repo, clone_url
 
 
+def do_git_checkout(clone_url, checkout_path, github_reference):
+    """
+    Handle making a clone, setting a branch/tag and
+    enumerating the last commit for a role.
+
+    :param clone_url:
+        A valid anonymous/public github clone url.
+    :param checkout_path:
+        Where to make the clone.
+    :param github_reference:
+        A valid branch or a tag name.
+
+    :return: A tuple with the following members:
+        - the git.Repo object
+        - the enumerated github_reference (aka branch or tag)
+        - the last commit object from the github_reference or $HEAD
+    :rtype: tuple(git.Repo, string, git.Commit)
+    """
+    logger.info(f'CLONING {clone_url}')
+
+    # pygit didn't have an obvious way to prevent interactive clones ...
+    pid = subprocess.run(
+        f'GIT_TERMINAL_PROMPT=0 git clone --recurse-submodules {clone_url} {checkout_path}',
```

Comment:
It looks like this code was copied from elsewhere in the file, but this type of string interpolation with `shell=True` is really not a good idea...